controller/cmd/destination/main.go

```diff
@@ -41,7 +44,27 @@ func main() {
 		log.Fatalf("Failed to listen on %s: %s", *addr, err)
 	}
 
-	server, err := destination.NewServer(*addr, *k8sDNSZone, *controllerNamespace, *enableH2Upgrade, k8sAPI, done)
+	trustDomain := ""
+	if *disableIdentity {
+		log.Info("Identity is disabled")
+	} else {
+		global, err := config.Global(consts.MountPathGlobalConfig)
```

Comment:
When I test this branch, the destination container fails to start, since it's not configured to volume mount the linkerd-config ConfigMap in the install YAML. This fixes it:

```diff
diff --git a/chart/templates/controller.yaml b/chart/templates/controller.yaml
index 7caf7f56..11a96ad2 100644
--- a/chart/templates/controller.yaml
+++ b/chart/templates/controller.yaml
@@ -136,6 +136,9 @@ spec:
           containerPort: 8086
         - name: admin-http
           containerPort: 9996
+        volumeMounts:
+        - name: config
+          mountPath: /var/run/linkerd/config
         image: {{.Values.ControllerImage}}
         imagePullPolicy: {{.Values.ImagePullPolicy}}
         args:
```